### ISSUE TITLE
Catch rejected promises in shell scripts

### ISSUE THREAD

**Issue by LoicPoullain** [FoalTS/foal#671]
When an error is thrown in an `async main` of a shell script, we get this warning:

```
(node:15576) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:15576) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

We can probably fix this by making `run-script` async in `packages/cli/src/run-script/run-script.ts` and wrap up `main(args)` with a `trycatch (err) { console.log(err) }`

**Comment by damlys** [comment]
Can be improved with following event handlers :ok_hand: 

- https://nodejs.org/api/process.html#process_event_uncaughtexception
- https://nodejs.org/api/process.html#process_event_unhandledrejection

```typescript
process.on("uncaughtException", (error: any): void => {
  process.stderr.write(`[uncaught_exception] ${error.toString()}\n`);
  process.exit(1);
});

process.on("unhandledRejection", (reason: any, promise: Promise<any>): void => {
  process.stderr.write(`[unhandled_rejection] ${reason.toString()}\n`);
  process.exit(1);
});
```

**Comment by LoicPoullain** [comment]
I'd rather not to go with global event handlers because:
- they are hard to unit test
- and they may hide unexpected errors thrown by the function `runScript` and not the script itself.

I think that a try/catch is more appropriate in this case.

**Comment by LoicPoullain** [comment]
Fixed in v1.10.0
